fsspec/asyn.py

```diff
@@ -369,12 +369,16 @@ async def _put(self, lpath, rpath, recursive=False, **kwargs):
             lpath = make_path_posix(lpath)
         fs = LocalFileSystem()
         lpaths = fs.expand_path(lpath, recursive=recursive)
-        rpaths = other_paths(lpaths, rpath)
+        dirs = [l for l in lpaths if os.path.isdir(l)]
+        rdirs = other_paths(dirs, rpath)
+        await _throttled_gather([self._makedirs(d, exist_ok=True) for d in rdirs])
```

Comment:
Since `makedirs()` doesn't do anything relevant with fds, I think it is safe to just use `asyncio.gather` for this.

Reply:
Is there any cost to always using _throttled (or whatever we finally call it) ?

Reply:
We will add callback management to it, so it is better just avoid using it if not necessary (and in this case, it isnt).